db.go

```diff
@@ -282,6 +285,17 @@ type Result interface {
 	// Where() discards the initial filtering conditions and sets new ones.
 	Where(...interface{}) Result
 
+	Join(...interface{}) Result
```

Comment:
I think the QueryBuilder, instead of returning a Result, should return a `Query` type